Sync Load Panic: invalid memory address or nil pointer dereference

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

We found some panic logs from some clusters:
```log

[2025/06/05 03:32:38.963 +00:00] [ERROR] [handle_hist.go:279] ["handleOneItemTask panicked"] [recover="runtime error: invalid memory address or nil pointer dereference"] [stack="github.com/pingcap/tidb/pkg/statistics/handle.(*Handle).handleOneItemTask.func1\n\t/workspace/source/tidb/pkg/statistics/handle/handle_hist.go:279\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:914\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:261\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:861\ngithub.com/pingcap/tidb/pkg/statistics/handle.(*Handle).readStatsForOneItem\n\t/workspace/source/tidb/pkg/statistics/handle/handle_hist.go:376\ngithub.com/pingcap/tidb/pkg/statistics/handle.(*Handle).handleOneItemTask\n\t/workspace/source/tidb/pkg/statistics/handle/handle_hist.go:327\ngithub.com/pingcap/tidb/pkg/statistics/handle.(*Handle).HandleOneTask\n\t/workspace/source/tidb/pkg/statistics/handle/handle_hist.go:251\ngithub.com/pingcap/tidb/pkg/statistics/handle.(*Handle).SubLoadWorker\n\t/workspace/source/tidb/pkg/statistics/handle/handle_hist.go:210"]
--

```

### 2. What did you expect to see? (Required)

No nil pointer panic.

### 3. What did you see instead (Required)
Panics

### 4. What is your TiDB version? (Required)

7.5.6

--- a/pkg/statistics/handle/handle_hist.go
+++ b/pkg/statistics/handle/handle_hist.go
@@ -309,8 +309,15 @@ func (h *Handle) handleOneItemTask(task *NeededItemTask) (err error) {
 		wrapper.idx = index
 	} else {
 		col, ok := tbl.Columns[item.ID]
-		if !ok || col.IsFullLoad() {
-			wrapper.col = nil
+		if !ok {
+			// From `removeHistLoadedColumns`, we can see that if a column is not found in the `tbl`,
+			// we'll directly remove it from the `remainedItems`.
+			// This precondition makes it difficult to tell when we will get here.
+			// But we should not panic here.
+			return nil
+		} else if col.IsFullLoad() {
+			// If this column is fully loaded, we don't need to load it again.
+			return nil
 		} else {
 			wrapper.col = col
 		}

